components/shared_code/src/shared_data_model/meta/data_model.py

```diff
@@ -55,15 +56,15 @@ def check_source_urls(self) -> None:
 
     def check_logos(self) -> None:
         """Check that a logo exists for each source and vice versa."""
-        logos_path = pathlib.Path(__file__).parent.parent / "logos"
         for source_type in self.sources:
-            logo_path = logos_path / f"{source_type}.png"
-            if not logo_path.exists():
-                msg = f"No logo exists for {source_type}"
-                raise ValueError(msg)
-        for logo_path in logos_path.glob("*.png"):
-            if logo_path.stem not in self.sources:
-                msg = f"No source exists for {logo_path}"
+            try:
+                get_logo(source_type)
```

Comment:
Reading the binary seems like a rather expensive operation; if we only want to check that the file exists, the entry in `get_logo_names()` should suffice?

Reply:
Yes, changed.